org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServers.java

```diff
@@ -268,6 +269,42 @@ public long getStartVersion() {
 			return this.startVersion;
 		}
 
+		public SingleServerDocumentExecutor toExecutor(final @NonNull LanguageServerWrapper serverWrapper) {
+			return new SingleServerDocumentExecutor(this.document, serverWrapper);
+		}
+	}
+
+	/**
+	 * Executor for a single server - for follow-up requests that should go to the server that returned an earlier response
+	 *
+	 */
+	public static class SingleServerDocumentExecutor extends LanguageServerDocumentExecutor {
+
+		private final @NonNull LanguageServerWrapper serverWrapper;
+
+		SingleServerDocumentExecutor(final @NonNull IDocument document, final @NonNull LanguageServerWrapper serverWrapper) {
```

Comment:
the LanguageServerDocumentExecutor uses the document to find out the wrappers and connect them if needed, but I do not think this executor uses the document for anything.

Reply:
You are (partly) right - one of the motivations for this class was as a replacement for `DocumentInfo`, which effectively bundles a reference to a single LS with a document - it's virtually a POJO with no brains. In some of the code I was trying to port, the existing logic passed through a `DocumentInfo` as a single field. My first attempt replaced this with passing around a `LanguageServerWrapper` and an `IDocument` as individual arguments. But then I thought of taking your `LanguageServerWrapperExecutor` idea and marrying it with `LanguageServerDocumentExecutor`.

I think this gives a nicer way of rewriting the existing code, and also means versioning support is available.